examples/grover_algorithm_tutorial.ipynb

```diff
@@ -30,248 +29,505 @@
     "import sympy as sp\n",
     "import matplotlib.pyplot as plt\n",
     "import matplotlib as mpl\n",
-    "from hierarqcal import Qinit, Qmask, Qunitary, Qpivot, Qcycle, Qunmask\n",
+    "from hierarqcal import Qinit, Qmask, Qunmask, Qmotif, Qunitary, Qpivot, Qcycle\n",
     "\n",
-    "mpl_backend = mpl.get_backend()"
-   ]
-  },
-  {
-   "attachments": {},
-   "cell_type": "markdown",
-   "id": "cca00e17",
-   "metadata": {},
-   "source": [
-    "## Background\n"
+    "mpl_backend = mpl.get_backend()\n"
    ]
   },
   {
    "cell_type": "markdown",
    "id": "fca51beb",
    "metadata": {},
    "source": [
-    "Grover's algorithm provides an optimal method to amplify the amplitude of a specific state using an oracle. This specific state is usually referred to as the *target state* which we label  $T$. The oracle is a \"black-box\" with the ability to change the sign of amplitude of the target state, in other words add a $\\pi$ phase to the amplitude of the target state. This processes is commonly described as *marking* the target. \n",
+    "## Background\n",
+    "\n",
+    "Imagine someone hands you a definition of a word. They then ask you to use a dictionary to find the word that has a definition that exactly matches the given definition. If your dictionary contains $N$ words, on average you will need to check (query) $N/2$ times whether a word's definition exactly matches the given definition before you can expect to find an exact match. This is intuitively correct since sometimes you will be lucky and find the word immediately, while other times you will need to search through the entire dictionary before you find the word you seek. \n",
+    "\n",
+    "This is a typical example of a search of an unstructured database. In our example a dictionary, while clearly structured alphabetically, is not structured with respect to the definitions it contains. Grover's algorithm is a quantum algorithm that provides a speed-up with respect to the number of steps needed, allowing one to find the sought entry in order $\\sqrt{N}$ queries.\n",
+    "\n",
+    "### The algorithm\n",
+    "\n",
+    "Grover's algorithm provides an optimal method to amplify the amplitude of a specific state. This specific state is usually referred to as the *target state* which we label  $T$. The algorithm requires an oracle, that is a \"black-box\" with the ability to change the sign of the amplitude of the target state. This processes is commonly described as *marking* the target. \n",
     "\n",
     "Grover's algorithm works in two steps:\n",
     "\n",
     "Step 1: Apply the oracle. \n",
```

Comment:
Also minor can you make Step 1 and Step 2 bold, i.e. 
**Step 1:** 
**Step 2:**